pkg/network/controllers/vm.go

```diff
@@ -68,11 +68,22 @@ func NewVMController(clientset kubevirt.Interface) *VMController {
 }
 
 func (v *VMController) Sync(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance) (*v1.VirtualMachine, error) {
-	if vmi == nil || vmi.DeletionTimestamp != nil {
+	if vmi != nil && vmi.DeletionTimestamp != nil {
 		return vm, nil
 	}
 
 	vmCopy := vm.DeepCopy()
+	if vmi == nil {
+		ifaces, networks := ClearDetachedInterfaces(
+			vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces,
+			vmCopy.Spec.Template.Spec.Networks, nil,
+		)
+		vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces = ifaces
+		vmCopy.Spec.Template.Spec.Networks = networks
+
+		return vmCopy, nil
+	}
```

Comment:
Any reason for duplicating the VM mutation code?

If the `indexedStatusIfaces` is conditionally generated, we could keep the existing logic as-is.
```go
var indexedStatusIfaces map[string]v1.VirtualMachineInstanceNetworkInterface

if vmi != nil {
	indexedStatusIfaces = vmispec.IndexInterfaceStatusByName(vmi.Status.Interfaces,
		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool { return true },
	)
}
```

Reply:
> Any reason for duplicating the VM mutation code?

just because if we create a function, it will mutate the input args,
in this case your idea is better as it will be less sensitive for changes, will change